research-software-directory/RSD-as-a-service
# Multiple fixes concerning data lost on token refresh

# Token refresh and accordion list fix

Closes #503
Closes #507 
Closes #508 
Closes #511 

Changes proposed in this pull request:
*  The accordion title is now clearly visible for the longer items list (#507)
*  The edit software pages (sections) will not reload directly on the token refresh, which leads to lost of edited but not yet saved  data (#503)
*  The edit project pages (sections) will not reload directly on the token refresh, which leads to lost of edited but not yet saved  data (#508)
*  The organisation settings will not reload on token refresh (#511)

How to test:
* `docker-compose build frontend ` to build frontend image
* `docker-compose up` to start RSD
*  1st approach to test:
    * create new project and visit edit project page. Change some values and leave page open for 60min. When you are back your data should still be present and you should be able to save your changes.
    * in the console.log you will see message after token refresh is performed:
        * `waitInMs` will log when next token refresh will take a place (time you need to wait)
* 2nd approach to test:
   * add following env variable to `frontend/.env.local`: NEXT_PUBLIC_REFRESH_MARGIN_MSEC=3540000
   * start frontend in dev mode using `yarn dev`
   * visit RSD on http://localhost:3000
   * create new project and go to edit. The token refresh will be performed every 60 sec.
   * make changes and do not save them. Confirm that your changes are present after 60 seconds.
   * examine console logs, as described in the 1st approach

PR Checklist:
